### PR TITLE
Format void tag

### PR DESCRIPTION
**Pull request by mickm3n**
## Issue
Fixes https://github.com/maxhumber/gazpacho/issues/26

## Problem
string_to_dom is unable to handle void tags

## Solution
transform void tags to self-closing tags and transform back

## Changes
- Add test cases
- Use regex to find void tags
- Use a special random attribute as a marker to know which void tags have been transformed.

### CODE REVIEW

**Review by maxhumber — COMMENTED** [review]
Solid PR... just a couple of changes needed. Really appreciate your work so far!

File: gazpacho/utils.py [hunk 1]
```diff
@@ -53,9 +57,15 @@ def format(html: str, fail: bool = False) -> str:
     ```
     """
     try:
-        dom = string_to_dom(html)
+        # attribute placeholder as a mark to transform back
+        while True:        
+            placeholder = ''.join(choice(ascii_letters) for i in range(8))
```
[the remark this answers is not in the document]
Comment (mickm3n) [reply gazpacho/utils.py]: That will introduce another issue: attribute name should start with letters.
Do you have any other ideas?

File: gazpacho/utils.py [hunk 2]
```diff
@@ -53,9 +57,15 @@ def format(html: str, fail: bool = False) -> str:
     ```
     """
     try:
-        dom = string_to_dom(html)
+        # attribute placeholder as a mark to transform back
+        while True:
```
[the remark this answers is not in the document]
Comment (mickm3n) [reply gazpacho/utils.py]: Thanks for the comments. It should be a problem.
The error handling logic is a little messy, so I move to another function. Please help to review it again.

File: gazpacho/utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxhumber) [reply gazpacho/utils.py]: Solid point! 😬

File: gazpacho/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (maxhumber) [reply gazpacho/utils.py]: This is really solid work @mickm3n. I'm going to merge and credit you in the next release~

**Pull request merged**